### PR TITLE
Fix newsletter forms

### LINKED ISSUE #3628: Localize newsletter forms
Ensure that newsletter signups are in the correct locale and country.

Task: Replicate a similar form & features from https://www.mozilla.org/fr/newsletter/mozilla/
(Critical) Select country + language in dropdown lists
(Good to have) Select default language in the list based on the page locale
As things stand, people subscribing from a localized page are being subscribed to the EN list.
Not just from the footer form, but also when checking “Yes, I want to receive email updates about Mozilla’s campaigns.” from localized petitions.

@alanmoo to add more info

### PR DESCRIPTION
**Pull request by youriwims**
Closes #3628

Fix that sends snippet `country` and `lang` to basket.

### CODE REVIEW

File: network-api/networkapi/campaign/views.py [hunk 1]
```diff
@@ -125,13 +126,13 @@ def petition_submission(request, petition):
         "email": request.data['email'],
         "email_subscription": request.data['newsletterSignup'],
         "source_url": request.data['source'],
-        "lang": request.data['lang'],
+        "lang": request.data['lang']
```
Comment (Pomax) [inline network-api/networkapi/campaign/views.py]: this is python. Ending the last element on a comma is actually a good thing.

File: source/js/components/join/join.jsx [hunk 2]
```diff
@@ -311,6 +317,14 @@ export default class JoinUs extends React.Component {
    * Render localization fields
    */
 
+  setCountry(country) {
+    this.setState({ country: country });
```
Comment (Pomax) [inline source/js/components/join/join.jsx]: not technically wrong or right, but `{ country: country }` and `{ country }` are equivalent in modern JS.

File: network-api/networkapi/campaign/views.py (hunk 1 above)
Comment (youriwims) [reply network-api/networkapi/campaign/views.py]: Sorry about that—I'll revert it.

Comment (youriwims) [reply network-api/networkapi/campaign/views.py]: Should there have been errors raised for this when I ran `inv test`?

Comment (Pomax) [reply network-api/networkapi/campaign/views.py]: nope, both are fine, but leaving that trailing comma on is "the finer of the two" as far as python is concerned.

File: network-api/networkapi/campaign/views.py [hunk 3]
```diff
@@ -125,13 +126,13 @@ def petition_submission(request, petition):
...
     }
 
-    if petition.requires_country_code:
+    if petition:
         if 'country' in request.data:
             data["country"] = request.data['country']
-        else:
+        if petition.requires_country_code and 'country' not in request.data:
```
[the remark this answers is not in the document]
Comment (youriwims) [reply network-api/networkapi/campaign/views.py]: ah true.

**Comment by TheoChevalier** [comment]
Thank you both! I can confirm I can see both fields being sent in the request now 👍

There’s a small regression though, updating the country field sets the language to English.

STR: 
- open https://foundation-mofostaging-pr-3766.herokuapp.com/fr/
- click on the form in the footer
- Pick a language (German, for instance)
- Now pick a country

Current: Language is now set as English
Expected: Language remain German

**Pull request merged**